src/pages/Mypage/Mytaste/Mytaste.js

```diff
@@ -1,13 +1,185 @@
 import React, { Component } from 'react';
 import Nav from '../../../components/Nav/Nav';
+import Chart from './Chart/Chart';
+import PreferredCountryGenre from './PreferredCountreNation/PreferredCountryGenre';
+import WordCloud from './wordCloud/wordCloud';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faUserCog } from '@fortawesome/free-solid-svg-icons';
+
+import {
+  PREFERRED_API,
+  PREFERRED_TOKEN,
+  MYSTAR_API,
+  MYSTAR_TOKEN,
+} from '../../../config';
 import './mytaste.scss';
 
 class Mytaste extends Component {
+  constructor() {
+    super();
+    this.state = {
+      userData: {},
+      chartData: {
+        labels: ['', '1', '', '2', '', '3', '', '4', '', '5'],
+        datasets: [
+          {
+            label: 'Star Ratings',
+            barThickness: 35,
+            data: [1, 2, 3, 4, 9, 5, 3, 2, 1, 4],
+            backgroundColor: Array(10).fill('#fbdd62'),
+          },
+        ],
+      },
+      myStar: {},
+    };
+  }
+
+  componentDidMount() {
+    this.loadMystarData();
+    // this.changeColorChart();
+    this.loadPreferredData();
+  }
+
+  loadPreferredData = () => {
+    fetch(PREFERRED_API, {
+      method: 'GET',
+      headers: {
+        Authorization: PREFERRED_TOKEN,
+      },
+    })
+      .then((res) => res.json())
+      .then((res) => this.setState({ userData: res }))
+      .catch((error) => console.log('error', error));
+  };
+
+  loadMystarData = () => {
+    fetch(MYSTAR_API, {
+      method: 'GET',
+      headers: {
+        Authorization: MYSTAR_TOKEN,
+      },
+    })
+      .then((res) => res.json())
+      .then((res) =>
+        this.setState({ myStar: res.user }, () => {
+          this.setMyStar();
+        })
+      )
+      .catch((error) => console.log('error', error));
+  };
+
+  setMyStar = () => {
+    const { myStar, chartData } = this.state;
+    let tempData = [];
+    for (let key in myStar) {
+      tempData.push(myStar[key]);
+    }
+    chartData.datasets[0].data = tempData;
+    this.changeColorChart(chartData);
+  };
+
+  changeColorChart = (chartData) => {
+    // console.log(chartData);
+    const data = chartData.datasets[0].data;
+    const backgroundColor = this.state.chartData.datasets[0].backgroundColor;
```

Comment:
```suggestion
    const { data } = chartData.datasets[0]
    const { backgroundColor } = this.state.chartData.datasets[0]
```
구조분해 할당 사용해줄수도 있는데, state에서 가져온 값과 인자로 넘어온 값이 서로 달라야 하나요?
그렇지 않다면 다음과 같이 해줄 수 있겠네요.
```js
const { data, backgroundColor } = this.state.chartData.datasets[0]
```

Reply:
네 종택님~ 확인해서 수정했습니다 ^^